compiler.pri

```diff
@@ -227,39 +227,61 @@ unix {
 		QMAKE_OBJECTIVE_CXXFLAGS *= -O3 -march=native -ffast-math -ftree-vectorize -fprofile-use
 	}
 
-	CONFIG(c++11) {
-		# Qt 5 will pass the expected compiler flags
-		# needed for C++11 mode when CONFIG includes c++11.
-		# But Qt 4 won't, so we add it manually.
-		lessThan(QT_MAJOR_VERSION, 5) {
-			QMAKE_CXXFLAGS *= -std=c++11
+	# First, auto-populate CONFIG with the latest C++ standard
+	# that Qt has detected support for.
+	!CONFIG(c++03):!CONFIG(c++11):!CONFIG(c++14):!CONFIG(c++1z) {
+		contains(QT_CONFIG, c++11): CONFIG *= c++11
+		contains(QT_CONFIG, c++14): CONFIG *= c++14
+		contains(QT_CONFIG, c++1z): CONFIG *= c++1z
+	}
+	# If no C++ standard was auto-detected, choose C++03.
+	!CONFIG(c++03):!CONFIG(c++11):!CONFIG(c++14):!CONFIG(c++1z) {
+		CONFIG += c++03
+	}
+
+	# Qt 4, as opposed to Qt 5, doesn't automatically
+	# populate QMAKE_CXXFLAGS and QMAKE_LFLAGS with the
+	# appproiate -std=c++XX flags. So, for Qt 4, we'll
+	# have to set them up manually.
+	lessThan(QT_MAJOR_VERSION, 5) {
```

Comment:
I'm pretty sure I checked that this was the case.
They should be populated via the default_post.prf qmake feature file.
See: https://github.com/qt/qtbase/blob/1e97037ed97aade9eb107bbc595785c2de76f7dc/mkspecs/features/default_post.prf#L98

But I'll double-check.